### ISSUE TITLE
`reincarnate()` fails when prometheus client is installed but not configured

### ISSUE THREAD

**Issue by heidi-luke** [django-q2/django-q2#220]
*Problem*
We've run into an issue where a prometheus client is pulled in from a common library as a dependency but not used in-project, so doesn't have the `PROMETHEUS_MULTIPROC_DIR` environment variable set. This causes [this line](https://github.com/django-q2/django-q2/blob/2c456cd0aa05586ce9a85e370bd53d1cefd50a1a/django_q/cluster.py#L234) to fail when reincarnating a process with the following trace:

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/multiprocessing/process.py", line 314, in _bootstrap
    self.run()
  File "/usr/local/lib/python3.11/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/python_env/.d_venv/lib/python3.11/site-packages/django_q/cluster.py", line 165, in __init__
    self.start()
  File "/python_env/.d_venv/lib/python3.11/site-packages/django_q/cluster.py", line 174, in start
    self.guard()
  File "/python_env/.d_venv/lib/python3.11/site-packages/django_q/cluster.py", line 315, in guard
    self.reincarnate(p)
  File "/python_env/.d_venv/lib/python3.11/site-packages/django_q/cluster.py", line 234, in reincarnate
    prometheus_multiprocess.mark_process_dead(process.pid)
  File "/python_env/.d_venv/lib/python3.11/site-packages/prometheus_client/multiprocess.py", line 169, in mark_process_dead
    for f in glob.glob(os.path.join(path, f'gauge_{mode}_{pid}.db')):
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen posixpath>", line 76, in join
        TypeError: expected str, bytes or os.PathLike object, not NoneType
```

This results in enqueued tasks getting stuck in a `pending` state and not getting processed until the service is rebooted.

*Workaround*
Adding `PROMETHEUS_MULTIPROC_DIR` as a dummy env variable with a value of something like `/tmp/metrics` seems to solve the issue but ideally there would be a check on a configured vs. non-configured prometheus client before running the code to mark the process dead.

**Comment by paulocoutinhox** [comment]
Hi,

There is a fix for this in the project?

**Comment by GDay** [comment]
Not yet, feel free to create a PR if you want to. I won't have time in the next few days at the very least unfortunately.

**Comment by paulocoutinhox** [comment]
I can. Can you only show me where is the problem about, in what file? Thanks.

**Comment by GDay** [comment]
Great, thanks.

https://github.com/django-q2/django-q2/blob/337a5782d9113a6e9d058743488e7517aebb8280/django_q/cluster.py#L233-L234

We would need an additional check there if `prometheus_client` has been setup or not. I haven't looked into it, but that seems like the most reasonable approach. 

Currently, if that library has been installed, but not configured, it will raise an error when `mark_process_dead` is being called.

**Comment by paulocoutinhox** [comment]
I made a PR: https://github.com/django-q2/django-q2/pull/231

**Comment by GDay** [comment]
Should be fixed, thanks to @paulocoutinhox!

**Comment by paulocoutinhox** [comment]
Can you create a new version?

**Comment by GDay** [comment]
Yeah, I will create one tomorrow. Want to run a few more manual tests before I release the new version.

**Comment by paulocoutinhox** [comment]
Hi, do you have time to release a new version? Thanks man.

**Comment by GDay** [comment]
Ah sorry, forgot. Will do later today.

**Comment by GDay** [comment]
Released v1.7.3
